### ISSUE TITLE
find_zeros error with Unitful

### ISSUE THREAD

**Issue by hsugawa8651** [JuliaMath/Roots.jl#434]
https://github.com/JuliaMath/Roots.jl/blob/4263c7683423af209af1879bd9cd223b0ba55acd/src/find_zeros.jl#L110

`find_zeros` fails when it calls `approx_close`.
This is due to the dimension mismatch between two values calculated by this line.

```julia
using Unitful
using Roots
xzs1=find_zeros(x -> cos(x / 1u"m"), -1.6u"m",2u"m")

DimensionError: 2.3406689268274554e-8 m and 5.477420592293902e-7 m¹ᐟ² are not dimensionally compatible.

Stacktrace:
 [1] _isless
   @ ~/.julia/packages/Unitful/GYzMo/src/quantities.jl:243 [inlined]
 [2] isless(x::Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}, y::Quantity{Float64, 𝐋¹ᐟ², Unitful.FreeUnits{(m¹ᐟ²,), 𝐋¹ᐟ², nothing}})
   @ Unitful ~/.julia/packages/Unitful/GYzMo/src/quantities.jl:233
 [3] max(x::Quantity{Float64, 𝐋¹ᐟ², Unitful.FreeUnits{(m¹ᐟ²,), 𝐋¹ᐟ², nothing}}, y::Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}})
   @ Base ./operators.jl:481
 [4] approx_close(z1::Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}, z2::Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}, xatol::Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}, xrtol::Float64)
   @ Roots ~/.julia/packages/Roots/neTBD/src/find_zeros.jl:110
 [5] find_zeros(f::var"#145#146", a::Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}, b::Quantity{Int64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}; no_pts::Int64, k::Int64, naive::Bool, kwargs::@Kwargs{})
   @ Roots ~/.julia/packages/Roots/neTBD/src/find_zeros.jl:387
 [6] find_zeros(f::Function, a::Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}, b::Quantity{Int64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}})
   @ Roots ~/.julia/packages/Roots/neTBD/src/find_zeros.jl:310
 [7] top-level scope
   @ In[172]:3
```

I wonder `approx_close()` may be substituted by `Base.isapprox()`  which works well with Unitful.
[function isapprox(x::Number, y::Number;](https://github.com/JuliaLang/julia/blob/680c7e3869a0d9da494bdface2efe058c2987b98/base/floatfuncs.jl#L220)

Namely, `approx_close(z1, z2, xatol, xrtol)`  may be equivalent to `isapprox(z1, z2; atol=xatol, rtol=xrtol)`

**Comment by jverzani** [comment]
Thanks for the report. Let me look at your suggestion and address this issue.

**Comment by jverzani** [comment]
Closed by #435
